test/integration/user/test/viewportHistogram.test.js

```diff
@@ -0,0 +1,154 @@
+            coordinates: [0, 0]
+        },
+        properties: {
+            id: 1,
+            value: 10,
+            category: 'a'
+        }
+    };
+    const feature2 = {
+        type: 'Feature',
+        geometry: {
+            type: 'Point',
+            coordinates: [10, 12]
+        },
+        properties: {
+            id: 2,
+            value: 1000,
+            category: 'a'
+        }
+    };
+    const feature3 = {
+        type: 'Feature',
+        geometry: {
+            type: 'Point',
+            coordinates: [10, 12]
+        },
+        properties: {
+            id: 3,
+            value: 1000,
+            category: 'b'
+        }
+    };
+    const featureWithNullCategory = {
+        type: 'Feature',
+        geometry: {
+            type: 'Point',
+            coordinates: [10, 12]
+        },
+        properties: {
+            id: 4,
+            value: 1000,
+            category: null
+        }
+    };
+    const featureWithEmptyStringCategory = {
+        type: 'Feature',
+        geometry: {
+            type: 'Point',
+            coordinates: [10, 12]
+        },
+        properties: {
+            id: 5,
+            value: 1000,
+            category: ''
+        }
+    };
+    const featureWithUndefinedCategory = {
+        type: 'Feature',
+        geometry: {
+            type: 'Point',
+            coordinates: [10, 12]
+        },
+        properties: {
+            id: 6,
+            value: 1000
+        }
+    };
+
+    function createMapWith (stringViz, features) {
+        const setup = util.createMap('map');
+        map = setup.map;
+        div = setup.div;
+
+        source = new carto.source.GeoJSON(features);
+        viz = new carto.Viz(stringViz);
+
+        layer = new carto.Layer('layer', source, viz);
+        layer.addTo(map);
+    }
+
+    const commonFeatures = {
+        type: 'FeatureCollection',
+        features: [feature1, feature2, feature3]
+    };
+
+    it('common use case', (done) => {
+        const histogramViz = '@histogram: viewportHistogram($category)';
+        createMapWith(histogramViz, commonFeatures);
+
+        layer.on('loaded', () => {
+            const histogram = viz.variables.histogram.value;
+            expect(histogram).toEqual([
+                { x: 'a', y: 2 },
+                { x: 'b', y: 1 }
+            ]);
+            done();
+        });
+    });
+
+    it('with nulls & undefined categories', (done) => {
+        const featuresWithSomeNulls = {
+            type: 'FeatureCollection',
+            features: [feature1, feature2, feature3,
+                featureWithNullCategory, featureWithEmptyStringCategory, featureWithUndefinedCategory]
+        };
+
+        const histogramViz = '@histogram: viewportHistogram($category)';
+        createMapWith(histogramViz, featuresWithSomeNulls);
+
+        layer.on('loaded', () => {
+            const histogram = viz.variables.histogram.value;
+            expect(histogram).toEqual([
```

Comment:
BTW, aren't we missing the _`undefined`_ feature here?

Reply:
That's what I was saying in my comments 🔝 
Both features are been counted under the 'null' category, which I think it's pretty fine.

Reply:
Hahaha. See my latest https://github.com/CartoDB/carto-vl/pull/1187#discussion_r240586723.

Sorry for the noise, but I wanted to understand better what was going on.